hazelcast/src/main/java/com/hazelcast/client/impl/protocol/template/ResponseTemplate.java

```diff
@@ -160,11 +160,17 @@ void Exception(int errorCode, String className, @Nullable String message, StackT
             , int causeErrorCode, @Nullable String causeClassName);
 
     /**
-     * @param readCount Number of items in the response.
+     * @param readCount The number of items read from the ringbuffer. This can be different from
+     *                  the size of the item array if a filter was applied when reading and some
+     *                  items were skipped and are not in the returned array. This can also be
+     *                  seen as the difference between the sequence ID of the last and first
+     *                  item in the returned array and the delta with which the user should
+     *                  increase his sequence counter to request the next set from the ringbuffer.
      * @param items     The array of serialized items.
+     * @param itemSeqs  sequence IDs of returned ringbuffer items
      */
     @Response(ResponseMessageConst.READ_RESULT_SET)
-    void ReadResultSet(int readCount, List<Data> items);
+    void ReadResultSet(int readCount, List<Data> items, @Since("1.5") @Nullable long[] itemSeqs);
```

Comment:
The extension seems to be ringbuffer specific. Extending the existing message will affect all users of READ_RESULT_SET. I would prefer designing a new response message like this:
        @Response(ResponseMessageConst.RINGBUFFER_READ_RESULT_SET)
        void ReadResultSet(int readCount, List<Map.Entry<Long, Data>> items);